tests/foreman/api/test_errata.py

```diff
@@ -88,123 +132,690 @@ def _validate_errata_counts(module_org, host, errata_type, expected_value, timeo
+        # searching back 1s prior to install.
+        target_sat.api_factory.wait_for_errata_applicability_task(
+            host_id=rhel_contenthost.nailgun_host.id,
+            from_when=epoch_timestamp,
+        )
+        # outdated package found on host
+        assert rhel_contenthost.run(f'rpm -q {package_filename}').status == 0
+        """
+            Modifying the applicable package:
+            - changed package applicability count by one and only one.
+            - changed errata applicability count by number of affected errata, whose
+                applicability status changed after package was modified.
+            - changed lists of applicable packages and applicable errata accordingly.
+        """
+        passed_checks = package_applicability_changed_as_expected(
+            target_sat,
+            rhel_contenthost,
+            package_filename,
+            prior_app_errata,
+            pre_errata_count,
+            pre_package_count,
+        )
+        # If passed_checks is False, this package was not applicable, continue to next.
+        if passed_checks is True:
+            present_applicable_packages.append(package_filename)
+
+    # Some applicable errata(s) now expected for outdated packages
+    assert rhel_contenthost.applicable_errata_count > 0
+    # Expected applicable package(s) now for the applicable errata
+    assert rhel_contenthost.applicable_package_count == len(present_applicable_packages)
+    post_app_errata = _fetch_available_errata_instances(target_sat, rhel_contenthost)
+    """Installing all YUM_9 security errata sequentially, if applicable.
+       after each install, applicable-errata-count should drop by one,
+       one or more of the erratum's listed packages should be updated.
+    """
+    installed_errata = []
+    updated_packages = []
+    expected_errata_to_install = [
+        errata.errata_id
+        for errata in post_app_errata
+        if errata.errata_id in FAKE_9_YUM_SECURITY_ERRATUM
+    ]
+    all_applicable_packages = set(
+        package for errata in post_app_errata for package in errata.packages
+    )
+    security_packages_to_install = set()
+    for errata_id in FAKE_9_YUM_SECURITY_ERRATUM:
+        errata_instance = (
+            target_sat.api.Errata().search(query={'search': f'errata_id="{errata_id}"'})[0].read()
+        )
+        present_packages_impacted_by_errata = [
+            package
+            for package in errata_instance.packages
+            if package in FAKE_9_YUM_UPDATED_PACKAGES
+        ]
+        security_packages_to_install.update(present_packages_impacted_by_errata)
+    # Are expected security errata packages found in all applicable packages ?
+    assert security_packages_to_install.issubset(all_applicable_packages)
+    # errata_instances = [target_sat.api.Errata().search(query={'search': f'errata_id="{_id}"'})[0].read() for _id in FAKE_9_YUM_SECURITY_ERRATUM]
```

Comment:
Is this forgotten debug variable?

Reply:
Thank you sir! Yes removed this in the latest push